Dockerfile

```diff
@@ -0,0 +1,65 @@
+FROM ubuntu:24.04
+
+ENV RUST_VERSION 1.74
+ENV MDBOOK_QUIZ_VERSION 0.3.3
+ENV MDBOOK_VERSION 0.4.34
+ENV AQUASCOPE_VERSION 0.3.1
+ENV AQUASCOPE_TOOLCHAIN nightly-2023-08-25
+ENV PATH "$PATH:/home/ubuntu/bin:/home/ubuntu/.volta/bin:/home/ubuntu/.cargo/bin"
+
+COPY . /opt/rust-book
+
+RUN echo \
+  && apt-get update \
+  && apt install -y rustup curl wget gcc git nginx \
+  && chown -R ubuntu /opt/rust-book
+
+USER ubuntu
```

Comment:
Why run all these commands as `ubuntu` rather than `root`?

Reply:
Mainly just to be the same uid as when the container runs and file permissions are thus properly set